contracts/Settlement/SettlementLogic.sol

```diff
@@ -446,52 +528,148 @@ contract SettlementLogic is ISettlement, SettlementStorage {
+                path,
+                minFeeAmount
             );
-            feeInRbtc = sovrynSwapNetwork.rateByPath(path, fee);
         }
-        require(
-            feeInRbtc > minFee,
-            "Order amount is too low to pay the relayer fee"
+
+        if (estOrderFee < minFeeAmountInToken) {
+            require(amountToFill > minFeeAmountInToken, "Order amount is too low to pay the relayer fee");
+            require(
+                orderSize > minFeeAmountInToken,
+                "Order amount is too low to pay the relayer fee"
+            );
+
+            relayerFee = minFeeAmountInToken;
+        } else {
+            uint256 fillAmountInRBtc = amountToFill;
+            uint256 minFillingAmount = isSpot
+                ? minSwapOrderSize
+                : minMarginOrderSize;
+            if (fromToken != WRBTC_ADDRESS) {
+                address[] memory pathToRbtc = sovrynSwapNetwork.conversionPath(
+                    fromToken,
+                    WRBTC_ADDRESS
+                );
+                fillAmountInRBtc = sovrynSwapNetwork.rateByPath(
+                    pathToRbtc,
+                    amountToFill
+                );
+            }
+
+            require(
+                fillAmountInRBtc >= minFillingAmount,
+                "Filling amount is too low"
+            );
+            relayerFee = estOrderFee;
+        }
+    }
+
+    function _depositOrderAsset(
+        address owner,
+        address assetAddress,
+        uint256 amount
+    ) internal {
+        if (assetAddress == WRBTC_ADDRESS) {
+            IWrbtcERC20 wrbtc = IWrbtcERC20(WRBTC_ADDRESS);
+            require(balanceOf[owner] >= amount, "insufficient-balance");
+            balanceOf[owner] -= amount;
+            wrbtc.deposit{value: amount}();
+        } else {
+            IERC20(assetAddress).safeTransferFrom(owner, address(this), amount);
+        }
+    }
+
+    function _checkWithdrawalOnCancel(
+        address owner,
+        address token,
+        uint256 amount
+    ) internal {
+        if (token == WRBTC_ADDRESS && balanceOf[owner] >= amount) {
+            withdraw(amount);
+        }
+    }
+
+    function _getPrice(uint256 amount, address fromToken, address toToken) internal returns (uint256 price) {
+        address[] memory _path = sovrynSwapNetwork.conversionPath(
+            fromToken,
+            toToken
         );
+        uint256 toAmount = sovrynSwapNetwork.rateByPath(_path, amount);
+        price = toAmount.mul(10**18).div(amount);
+    }
+
+    // Checks allowance of settlement contract for spending token,
+    // if allowance < needed, approve unlimited amount
+    function _checkAllowance(address tokenAdr, address spender, uint256 amount) internal {
+        IERC20 token = IERC20(tokenAdr);
+        uint256 allowance = token.allowance(address(this), spender);
+        if (allowance < amount) {
+            token.safeApprove(spender, UNLIMITED_ALLOWANCE);
```

Comment:
I am not sure if `approve` is the best viable option as it is susceptible to sandwich attacks. `safeIncreaseAllowance` in my opinion is a better option but given that we only increase this approval once, I think `safeApprove` should work fine as well. Keeping it open for discussion.

Reply:
because you're giving unlimited allowance anyway, i do not think you have to worry about sandwich attacks. i agree with you stan, that safeApproval shouln't be an issue in this specific case, but approve would still be safer here